tests/test_client.py

```diff
@@ -10,47 +10,57 @@
 
 ## Initialization.
 
+
+@pytest.mark.asyncio
 @with_client(invalid_kwarg=False)
 def test_client_superfluous_arguments(server, client):
     assert client.logger.warning.called
 
 
 ## Connection.
-
+@pytest.mark.asyncio
 @with_client()
-def test_client_reconnect(server, client):
-    client.disconnect(expected=True)
+async def test_client_reconnect(server, client):
+    await client.disconnect(expected=True)
     assert not client.connected
 
-    client.connect(reconnect=True)
+    await client.connect(reconnect=True)
```

Comment:
Im not convinced this test is enough, disconnection logic has been a particular trouble spot for this library.

If you reduce the disconnect timer on the mock client and sleep past it, we can add an assertion here that pydle didn't reconnect itself when it wasn't supposed to.

Reply:
Ultimately I lack the knowledge of the disconnection logic enough to make that change and have any idea what I'm doing, so I will leave that for a later coder to solve, unfortunately.

Reply:
fair enough, problem for a later time.